src/Build/Evaluation/IntrinsicFunctions.cs

```diff
@@ -480,6 +481,21 @@ internal static bool VersionLessThanOrEquals(string a, string b)
             return SimpleVersion.Parse(a) <= SimpleVersion.Parse(b);
         }
 
+        internal static string GetTargetFrameworkIdentifier(string tfm)
+        {
+            return NuGetFramework.Parse(tfm).Framework;
+        }
+
+        internal static string GetTargetFrameworkVersion(string tfm)
+        {
+            return NuGetFramework.Parse(tfm).Version.ToString(2);
+        }
+
+        internal static bool IsTargetFrameworkCompatible(string tfm1, string tfm2)
```

Comment:
I don't know I'm not familiar with `AssetTargetFallback`. What does that entail?

Reply:
[comment in code](https://github.com/microsoft/msbuild/blob/master/src/MSBuild/MSBuild/Microsoft.Build.CommonTypes.xsd#L1637)
[docs](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#assettargetfallback)

Reply:
[NuGetFrameworkUtility.IsCompatibleWithFallbackCheck](https://github.com/NuGet/NuGet.Client/blob/3803820961f4d61c06d07b179dab1d0439ec0d91/src/NuGet.Core/NuGet.Frameworks/NuGetFrameworkUtility.cs#L122) does check for compatibility with fallback TFMs. I'm not sure about AssetTargetFallback specifically but it makes sense to me to be consistent with the way NuGet determines compatibility.

Reply:
We'll have to check with NuGet folks about the shape of the API. I was envisioning something like `IsTargetFrameworkCompatible(candidate, ideal, fallback1, fallback2, fallback3)`. So you could do something like `IsTargetFrameworkCompatible($(thisprojecttfm), $(whatever), $(AssetTargetFallback)`?

But maybe this is a bad idea or unnecessary, not sure.

Reply:
IsCompatibleWithFallbackCheck does not consider ATF. 

We prefer that ATF remains a 2 methods approach because you need to ensure that it's only applied when necessary. 
Allows you to raise warnings and related better. 

We usually use NuGetFrameworkUtility.GetNearest for everything, but you probably just need to call IsCompatible. 

We basically do a 2 pass: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Build.Tasks/GetReferenceNearestTargetFrameworkTask.cs. 

While it does feel like it'd be nice to have an ATF intrinsic, what's a real scenario for that?

Reply:
Seems reasonable to go without until we find a concrete need.

@sfoslund did this get considered and rejected?

> you probably just need to call IsCompatible.

Reply:
No, I switched over to IsCompatible: https://github.com/microsoft/msbuild/pull/5234/files#diff-5af5289df4605c1a0a27f11cb43a8d23R41

Reply:
Ah, I was confused because the local name is still `IsCompatibleWithFallbackCheck`. Should that match?

Reply:
Yup, it does now :)